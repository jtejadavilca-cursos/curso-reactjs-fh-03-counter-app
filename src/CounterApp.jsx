import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState( value );

    const handleAdd = () => setCounter( counter + 1 );
    const handleSubs = () => setCounter( counter - 1 );
    const handleReset = () => setCounter( value );

  return (
    <>
        <h1>CounterApp</h1>
        <h1> { counter } </h1>

        <button onClick={ handleAdd }> +1 </button>
        <button onClick={ handleSubs }> -1 </button>
        <button onClick={ handleReset }> Reset </button>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: -1
}