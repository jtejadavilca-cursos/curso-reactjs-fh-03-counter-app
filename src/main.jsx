import React from 'react';
import ReactDOM from 'react-dom/client';

import { CounterApp } from './CounterApp';
// import { FirstApp } from './FirstApp';
//import { HelloWorldApp } from './HelloWorldApp';

import './css/styles.css'

ReactDOM.createRoot( document.getElementById( 'root' ) )
.render(
    <React.StrictMode>
        <CounterApp value={10} />

        {/* <FirstApp title="Hola, soy Goku" /> */}
    </React.StrictMode>
);