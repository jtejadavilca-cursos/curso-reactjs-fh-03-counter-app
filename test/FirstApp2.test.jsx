import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruyebas de <FirstApp />', () => {

    const title = 'Hola, soy Goku';
    const subTitle = 'soy un subtÃ­tulo';

    test('debe de hacer match con el snapshot', () => {

        const { container } = render(<FirstApp title={title}/>);
        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar elmensaje "Hola, soy Goku"', () => {
        render(<FirstApp title={title}/>);
        expect(screen.getByText( title )).toBeTruthy();
    });

    test('debe de mostrar el titulo en un H1', () => {
        render(<FirstApp title={title}/>);
        
        expect( screen.getByRole('heading', {level: 1}).innerHTML ).toContain( title );
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        render(<FirstApp title={title} subTitle={subTitle}/>);
        
        expect( screen.getAllByText(subTitle).length ).toBe( 2 );
    });
});