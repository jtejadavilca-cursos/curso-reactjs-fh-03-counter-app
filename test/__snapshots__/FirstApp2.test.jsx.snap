// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruyebas de <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
    No hay subTitle
  </p>
  <p>
    No hay subTitle
  </p>
  <p>
    Jose Tejada
  </p>
</div>
`;
